recipes/apr-util/all/conanfile.py

```diff
@@ -0,0 +1,204 @@
+        self._cmake.definitions["INSTALL_PDB"] = False
+        self._cmake.definitions["APU_HAVE_CRYPTO"] = self._with_crypto
+        self._cmake.definitions["APR_HAS_LDAP"] = self.options.with_ldap
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    @property
+    def _with_crypto(self):
+        return self.options.with_openssl or self.options.with_nss or self.options.with_commoncrypto
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        self._autotools.libs = []
+        self._autotools.include_paths = []
+        if self._with_crypto:
+            if self.settings.os == "Linux":
+                self._autotools.libs.append("dl")
+        my_unix_path = tools.unix_path if tools.os_info.is_windows else lambda x: x
+        conf_args = [
+            "--with-apr={}".format(my_unix_path(self.deps_cpp_info["apr"].rootpath)),
+            "--with-crypto" if self._with_crypto else "--without-crypto",
+            "--with-openssl={}".format(my_unix_path(self.deps_cpp_info["openssl"].rootpath)) if self.options.with_openssl else "--without-openssl",
+            "--with-expat={}".format(my_unix_path(self.deps_cpp_info["expat"].rootpath)) if self.options.with_expat else "--without-expat",
+            "--with-mysql={}".format(my_unix_path(self.deps_cpp_info["libmysqlclient"].rootpath)) if self.options.with_mysql else "--without-mysql",
+            "--with-pgsql={}".format(my_unix_path(self.deps_cpp_info["libpq"].rootpath)) if self.options.with_postgresql else "--without-pgsql",
+            "--with-sqlite3={}".format(my_unix_path(self.deps_cpp_info["sqlite3"].rootpath)) if self.options.with_sqlite3 else "--without-sqlite3",
+            "--with-ldap={}".format(my_unix_path(self.deps_cpp_info["ldap"].rootpath)) if self.options.with_ldap else "--without-ldap",
+            "--with-berkeley-db={}".format(my_unix_path(self.deps_cpp_info["libdb"].rootpath)) if self.options.dbm == "db" else "--without-berkeley-db",
+            "--with-gdbm={}".format(my_unix_path(self.deps_cpp_info["gdbm"].rootpath)) if self.options.dbm == "gdbm" else "--without-gdbm",
+            "--with-ndbm={}".format(my_unix_path(self.deps_cpp_info["ndbm"].rootpath)) if self.options.dbm == "ndbm" else "--without-ndbm",
+
+        ]
+        if self.options.dbm:
+            conf_args.append("--with-dbm={}".format(self.options.dbm))
+        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def build(self):
+        if self.options.shared != self.options["apr"].shared:
+            raise ConanInvalidConfiguration("apr-util must be built with same shared option as apr")
```

Comment:
We used to force it on configure method, but I prefer this way, it's more explicit and give to user the chance of make a decision